storage/store.go

```
@@ -535,11 +536,36 @@ func (s *Store) GetVaults() ([]*Vault, error) {
 
 // DeleteVault deletes a vault and its coins
 func (s *Store) DeleteVault(publicKeyECDSA string) error {
-	if _, err := s.db.Exec("DELETE FROM coins WHERE public_key_ecdsa = ?", publicKeyECDSA); err != nil {
+	// Every delete is one unit: a failure part-way through must not strip a
+	// vault of its keyshares while leaving the vault itself in place, because
+	// no retry can restore that key material.
+	tx, err := s.db.Begin()
```

golangci-lint (2.13.2)

[error] (*database/sql.DB).Begin must not be called. use (*database/sql.DB).BeginTx

(noctx)